### PR TITLE
RUMM-3194 add sampling rate on Tracer

### PR DESCRIPTION
**Pull request by xgouchet**
### What and why?

Allow customer sampling their traces on the mobile side

### How?

A brief description of implementation details of this PR.

### Review checklist
- [ ] Feature or bugfix MUST have appropriate tests (unit, integration)
- [ ] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: Tests/DatadogTests/Matchers/SpanMatcher.swift [hunk 1]
```diff
@@ -18,6 +20,9 @@ extension String: AllowedSpanAttributeValue {}
 extension UInt64: AllowedSpanAttributeValue {}
 extension Int: AllowedSpanAttributeValue {}
 
+// Only numeric values are allowed for `span.ç_dd.*`.
```
Comment (fuzzybinary) [inline Tests/DatadogTests/Matchers/SpanMatcher.swift:23]: Is that supposed to be `ç_dd` or just `_dd`

Comment (xgouchet) [reply Tests/DatadogTests/Matchers/SpanMatcher.swift:23]: Indeed thanks.

**Comment by xgouchet** [comment]
> Do spans affect their children? If so, do we need to check / test the spans created by the URLSession delegate, and make sure the sampled Spans get encoded to the W3C / b3 headers?

The trace's sampling is being handled by the root span, meaning that this is going to be handled automatically by APM's logic on DD servers

**Review by maxep — APPROVED** [review]
Nice, thanks @xgouchet!

**Pull request merged**